src/python/pants/backend/jvm/tasks/jvm_compile/rsc/rsc_compile.py

```diff
@@ -0,0 +1,603 @@
+    return wu.stdout.rstrip()
+  with open(wu.output_paths()['stdout']) as f:
+    return f.read().rstrip()
+
+
+class RscCompileContext(CompileContext):
+  def __init__(self,
+               target,
+               analysis_file,
+               classes_dir,
+               rsc_mjar_file,
+               jar_file,
+               log_dir,
+               zinc_args_file,
+               sources,
+               rsc_outline_dir,
+               outline_dir):
+    super(RscCompileContext, self).__init__(target, analysis_file, classes_dir, jar_file,
+                                               log_dir, zinc_args_file, sources)
+    self.rsc_mjar_file = rsc_mjar_file
+    self.rsc_outline_dir = rsc_outline_dir
+    self.outline_dir = outline_dir
+
+  def ensure_output_dirs_exist(self):
+    safe_mkdir(os.path.dirname(self.rsc_mjar_file))
+    safe_mkdir(self.rsc_outline_dir)
+    safe_mkdir(self.outline_dir)
+
+
+class RscCompile(ZincCompile):
+  """Compile Scala and Java code to classfiles using Rsc."""
+
+  _name = 'rsc_outline' # noqa
+
+  @classmethod
+  def implementation_version(cls):
+    return super(RscCompile, cls).implementation_version() + [('RscCompile', 7)]
+
+  @classmethod
+  def register_options(cls, register):
+    super(RscCompile, cls).register_options(register)
+
+    rsc_outline_version = '0.0.0-294-d7114447'
+    scalameta_version = '4.0.0-M10'
+
+    # TODO: it would be better to have a less adhoc approach to handling
+    #       optional dependencies. See: https://github.com/pantsbuild/pants/issues/6390
+    cls.register_jvm_tool(
+      register,
+      'workaround-metacp-dependency-classpath',
+      classpath=[
+        JarDependency(org = 'org.scala-lang', name = 'scala-compiler', rev = '2.11.12'),
+        JarDependency(org = 'org.scala-lang', name = 'scala-library', rev = '2.11.12'),
+        JarDependency(org = 'org.scala-lang', name = 'scala-reflect', rev = '2.11.12'),
+        JarDependency(org = 'org.scala-lang.modules', name = 'scala-partest_2.11', rev = '1.0.18'),
+        JarDependency(org = 'jline', name = 'jline', rev = '2.14.6'),
+        JarDependency(org = 'org.apache.commons', name = 'commons-lang3', rev = '3.3.2'),
+        JarDependency(org = 'org.apache.ant', name = 'ant', rev = '1.8.2'),
+        JarDependency(org = 'org.pegdown', name = 'pegdown', rev = '1.4.2'),
+        JarDependency(org = 'org.testng', name = 'testng', rev = '6.8.7'),
+        JarDependency(org = 'org.scalacheck', name = 'scalacheck_2.11', rev = '1.13.1'),
+        JarDependency(org = 'org.jmock', name = 'jmock-legacy', rev = '2.5.1'),
+        JarDependency(org = 'org.easymock', name = 'easymockclassextension', rev = '3.1'),
+        JarDependency(org = 'org.seleniumhq.selenium', name = 'selenium-java', rev = '2.35.0'),
+      ],
+      custom_rules=[
+        Shader.exclude_package('*', recursive=True),]
+    )
+    cls.register_jvm_tool(
+      register,
+      'rsc_outline',
```

Comment:
To follow up on the previous suggestion, this could be called `rsc_toolchain`.

Reply:
makes sense.

Reply:
Actually, I'm going to just call this rsc I think.